src/views/SignUpPage/components/RegionItem.tsx

```diff
@@ -1,48 +1,64 @@
+import { useRecoilState } from 'recoil';
 import { styled } from 'styled-components';
 
 import { IcCheckboxBlue, IcCheckboxGrey } from '../../@common/assets/icons';
+
+import { preferRegionState } from '@/recoil/atoms/signUpState';
 interface RegionItemProps {
   region: string;
-  isCheckedList: boolean[];
   index: number;
-  setIsCheckedList: React.Dispatch<React.SetStateAction<boolean[]>>;
   regionList: string[];
 }
 
 const RegionItem = (props: RegionItemProps) => {
-  const { region, isCheckedList, index, setIsCheckedList, regionList } = props;
+  const { region, index, regionList } = props;
+  const [isCheckedList, setIsCheckedList] = useRecoilState(preferRegionState);
 
   const handleCheck = () => {
-    const selectedCount = isCheckedList.filter((value) => value === true).length;
+    const selectedCount = isCheckedList.data.filter((value) => value === true).length;
     // 전체선택을 누르면
     if (index === 0) {
-      setIsCheckedList(() => {
+      setIsCheckedList((prev) => {
         //전체가 이미 선택돼있다면 전체 취소하기/ 선택돼있지 않다면 전체 선택하고 나머지 다 취소
-        const updatedList = !isCheckedList[0]
-          ? [true, ...Array(regionList.length - 1)]
-          : [false, ...Array(regionList.length - 1)];
-        return updatedList;
+        const isAllSelected = prev.data[0];
+        const updatedData = isAllSelected
+          ? [false, ...Array(regionList.length - 1)]
+          : [true, ...Array(regionList.length - 1)];
```

Comment:
```suggestion
        const updatedData = [...prev.data];
        updatedDate[0] = !prev.data[0];
```
다른 조건 분기에서 처럼 얕은복사해서 0 인덱스 값만 이렇게 바꿔줄 수 있겠죠 ? - ?